src/main/java/dev/dkvl/womutils/WomUtilsPlugin.java

```diff
@@ -205,6 +249,259 @@ public void onResponse(Call call, Response response)
 		});
 	}
 
+	private void sendPlayerRequest(Request request, String target, boolean add)
+	{
+		okHttpClient.newCall(request).enqueue(new Callback()
+		{
+			@Override
+			public void onFailure(Call call, IOException e)
+			{
+				log.warn("Error adding new member {}.", e.getMessage());
+			}
+
+			@Override
+			public void onResponse(Call call, Response response) {
+				try
+				{
+					buildErrorMessage(response, add);
+				}
+				catch (Exception e)
+				{
+					if (add)
+					{
+						groupMembers.add(target);
+					} else {
+						groupMembers.remove(target);
+					}
+				}
+				response.close();
+			}
+		});
+	}
+
+	private void buildErrorMessage(Response response, boolean add) throws IOException {
+		Type messageType = new TypeToken<JsonObject>() {}.getType();
+		String responseBody = response.body().string();
+		JsonObject data = gson.fromJson(responseBody, messageType);
+		String errorMessage = data.get("message").toString().replace("\"", "");
+
+		if (!add && errorMessage.contains("Successfully"))
+		{
+			return;
+		}
+
+		ChatMessageBuilder cmb = new ChatMessageBuilder();
+		cmb.append(ChatColorType.HIGHLIGHT).append(errorMessage);
+
+		chatMessageManager.queue(QueuedMessage.builder()
+				.type(ChatMessageType.CONSOLE)
+				.runeLiteFormattedMessage(cmb.build())
+				.build());
+	}
+
+	private void sendMembersRequest(Request request)
+	{
+		okHttpClient.newCall(request).enqueue(new Callback()
+		{
+			@Override
+			public void onFailure(Call call, IOException e)
+			{
+				log.warn("Error while fetching members list {}.", e.getMessage());
+			}
+
+			@Override
+			public void onResponse(Call call, Response response) throws IOException {
+				groupMembers.clear();
+				Type typeOfArrayList = new TypeToken<ArrayList<GroupMember>>() {}.getType();
+				String responseBody = response.body().string();
+				ArrayList<GroupMember> data = gson.fromJson(responseBody, typeOfArrayList);
+				response.close();
+
+				for (GroupMember gm : data) {
```

Comment:
This will happen on the okHttp thread and i am not sure it will race with something else, i think it should be ok though.